### PR TITLE
refactor(plugin/runner): Revise cache, module loading

### PR DESCRIPTION
**Pull request by kwonoj**
**Description:**

One of the oversight around design of `TransformExecutor` is encapsulating plugin module logic. It has access to the cache and do its own loading & storing. This means consumer of plugin runner have tricky challenge to control its caching system. First, there is no way to escape how swc_plugin_runner controls cache and cannot synchronize into their own, also depends on the usecases cannot control the features they want to opt in: for example, there's no way one interface uses in-memory cache, and another uses filesystem since it is compile time configured singleton.

PR revisits overall design of TransformExecutor: now it accepts a tratir `PluginModuleBytes`, which abstracts any kind of bytes we are dealing with, such as raw file slice or serialized `wasmer::Module`. Cache instantiation and managing is now bubbled up to the application level (`swc` in here), so if someone wants non-singleton caching or integrate into their own caching system it can be customized.

Lastly, deprecated `memory_cache` feature and only exposes `filesystem_cache`. Cache implementation uses in-memory is always available, and can opt in filesystem cache where it's supported.

**BREAKING CHANGE:**

This is clearly breaking changes for the consumers of swc_core. for the @swc/core, this PR takes care of necessary changes. I'll work on next-swc changes later once we have new @swc/core version with this changes.

### CODE REVIEW

File: crates/swc/src/plugin.rs [hunk 1]
```diff
@@ -165,50 +138,8 @@ impl RustPlugins {
...
-                            .with_context(|| {
-                                format!("failed to invoke `{}` as js transform plugin", &p.0)
-                            })?;
-                    }
-                }
-
-                serialized_program.deserialize().map(|v| v.into_inner())
-            },
-        )
+        // [TODO]: unimplemented
```
Comment (kdy1) [inline crates/swc/src/plugin.rs:141]: Is this fine?

Comment (kwonoj) [reply crates/swc/src/plugin.rs:141]: Yes, this execution path is for @swc/wasm only (https://github.com/swc-project/swc/pull/7408/files#diff-8987cc214ab0c7d63b833d31785ecb48d89d1bcc204d42508c26ecd72becc7b9L165) which never worked correctly, so taking out for now and revise once we have proper support.

File: crates/swc_plugin_runner/src/transform_executor.rs [hunk 2]
```diff
@@ -13,43 +13,365 @@ use swc_common::{
...
+    }
 }
 
+/// A struct encapsule executing a plugin's transform.
+pub struct TransformExecutor<T> {
+    source_map: Arc<SourceMap>,
+    unresolved_mark: swc_common::Mark,
+    metadata_context: Arc<TransformPluginMetadataContext>,
+    plugin_config: Option<serde_json::Value>,
+    module_bytes: T,
```
[the remark this answers is not in the document]
Comment (kwonoj) [reply crates/swc_plugin_runner/src/transform_executor.rs:175]: Good q, all I want is actually TransformExecutor to hold trait object PluginModuleBytes but I wasn't able to do so easily. Any way to make it without generics?

[the remark this answers is not in the document]
Comment (kdy1) [reply crates/swc_plugin_runner/src/transform_executor.rs:175]: I was thinking `module_bytes: Box<dyn PluginModuleBytes>`.

[the remark this answers is not in the document]
Comment (kwonoj) [reply crates/swc_plugin_runner/src/transform_executor.rs:175]: I tried and there was something odd to fail build, but maybe I'm doing wrong. Let me give another go.

**Review by kdy1 — APPROVED** [review]
swc-bump:
 - swc_plugin_runner

**Review by kdy1 — APPROVED** [review]
swc-bump:
 - swc_plugin_runner

**Review by kdy1 — CHANGES_REQUESTED** [review]
This PR makes CI for `swc_plugin_runner` extremely slow

**Comment by kwonoj** [comment]
> This PR makes CI for swc_plugin_runner extremely slow

 This change is supposed to not to make a big difference 🤔

**Comment by kwonoj** [comment]
Likely fixture setup issue since it no longer uses cache internally. let me try to check.

**Comment by kwonoj** [comment]
@kdy1 Ok, looks like back to normal.

i.e previous run https://github.com/swc-project/swc/actions/runs/5008644586/jobs/8976753580: 24m
this pr https://github.com/swc-project/swc/actions/runs/5010202295/jobs/8979868909: 24m

**Review by kdy1 — APPROVED** [review]

**Review by kdy1 — APPROVED** [review]

**Pull request merged**